zppy/templates/coupled_global.py

```diff
@@ -444,13 +519,24 @@ def plot(ax, xlim, exps, param_dict, rgn):
                         verbose=param_dict["verbose"],
                         vol=param_dict["vol"],
                     )
-
-    ax.set_ylim(get_ylim(param_dict["default_ylim"], extreme_values))
+    ylim = get_ylim(param_dict["default_ylim"], extreme_values)
+    print(f"ylim={ylim}")
+    ax.set_ylim(ylim)
     if param_dict["set_axhline"]:
         ax.axhline(y=param_dict["axhline_y"], lw=1, c="0.5")
     ax.set_title(param_dict["title"])
     ax.set_xlabel("Year")
-    ax.set_ylabel(param_dict["ylabel"])
+    units = param_dict["ylabel"]
+    print(units)
+    try:
+        c = callable(units)
+    except Exception as e:
+        print(e)
+        raise e
+    if c:
+        units = units(exps[0])  # How do we know which var's units to put as the ylabel?
```

Comment:
I had this commented a while ago, but the plots seem labeled with appropriate units. See [debug3 results](https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.forsyth2/zppy_test_debug3_www/pr-400v42/v2.LR.BGC-LNDATM.CONTRL/global_time_series/global_time_series_1870-1880_results/).

Note to self 3) I will try to look at the units with `ncdump` output tomorrow to see if they need fixing.

Reply:
```
$ cd /lcrc/group/e3sm/ac.sfeng1/E3SM_Simulations/20221127.v2.LR.BGC-LNDATM.CONTRL.ne30pg2_r05_EC30to60E2r2.chrysalis/archive/lnd/hist
$ ls | head -n 1
v2.LR.BGC-LNDATM.CONTRL.elm.h0.1870-01.nc
$ ncdump -h v2.LR.BGC-LNDATM.CONTRL.elm.h0.1870-01.nc | grep -E "(FLNT|TOTSOMC|TOTECOSYSC|NEE|TOTVEGC):units"
		NEE:units = "gC/m^2/s" ;
		TOTECOSYSC:units = "gC/m^2" ;
		TOTSOMC:units = "gC/m^2" ;
		TOTVEGC:units = "gC/m^2" ;
$ cd /lcrc/group/e3sm/ac.sfeng1/E3SM_Simulations/20221127.v2.LR.BGC-LNDATM.CONTRL.ne30pg2_r05_EC30to60E2r2.chrysalis/archive/atm/hist
$ ls | head -n 1
v2.LR.BGC-LNDATM.CONTRL.eam.h0.1870-01.nc
$ ncdump -h v2.LR.BGC-LNDATM.CONTRL.eam.h0.1870-01.nc | grep -E "(TS|FSNT|FLNT):units"
		FLNT:units = "W/m2" ;
		FSNT:units = "W/m2" ;
		TS:units = "K" ;
```

These units do match up with [debug 3 lnd](https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.forsyth2/zppy_test_debug3_www/pr-400v42/v2.LR.BGC-LNDATM.CONTRL/global_time_series/global_time_series_1870-1880_results/v2.LR.BGC-LNDATM.CONTRL_glb_lnd.pdf) and [debug3 atm](https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.forsyth2/zppy_test_debug3_www/pr-400v42/v2.LR.BGC-LNDATM.CONTRL/global_time_series/global_time_series_1870-1880_results/v2.LR.BGC-LNDATM.CONTRL_glb_atm.pdf).

Reply:
I think I just misunderstood the process when I made the comment `# How do we know which var's units to put as the ylabel?`, or I made it before an important code change.

For plotting a generic variable, we'd have `"ylabel": lambda exp: np.array(exp["annual"][var_name][1]),` set in `plot_generic` and thus:
```
    units = param_dict["ylabel"] # lambda exp: np.array(exp["annual"][var_name][1]),
    print(units)
    try:
        c = callable(units) # This is a function, so yes it's callable
    except Exception as e:
        print(e)
        raise e
    if c:
        units = units(exps[0]) # We'd apply the function to the first experiment in exps. (And exps only has one element in it).
        print(units)
    ax.set_ylabel(units)
```